src/transform/src/lib.rs

```diff
@@ -775,10 +798,10 @@ impl Optimizer {
     pub fn fast_path_optimizer(_ctx: &mut TransformCtx) -> Self {
         let transforms: Vec<Box<dyn Transform>> = vec![
             Box::new(canonicalization::ReduceScalars),
-            Box::new(literal_constraints::LiteralConstraints),
-            Box::new(canonicalize_mfp::CanonicalizeMfp),
+            Box::new(LiteralConstraints),
+            Box::new(CanonicalizeMfp),
             // We might have arrived at a constant, e.g., due to contradicting literal constraints.
-            Box::new(fold_constants::FoldConstants {
+            Box::new(FoldConstants {
```

Comment:
Do we not want the fixed point here?

Reply:
There are no Lets in the `fast_path_optimizer`, because this only runs on fast path peeks (as determined by `create_fast_path_plan`).

But we might need to alternate between `ReduceScalars` and `FoldConstants`, so now I've added a custom fixpoint loop that does just these two.

Reply:
Well, it seems ok to me that we have some special things in the `fast_path_optimizer`. If we were calling more general functions there, then I'd be a little bit afraid of possibly introducing optimizer performance regressions in the `fast_path_optimizer` when modifying those functions.